pkg/apis/application/v1alpha1/types.go

```diff
@@ -174,6 +174,8 @@ type ApplicationSourceHelm struct {
 	ReleaseName string `json:"releaseName,omitempty" protobuf:"bytes,3,opt,name=releaseName"`
 	// Values is Helm values, typically defined as a block
 	Values string `json:"values,omitempty" protobuf:"bytes,4,opt,name=values"`
+	// ParameterFiles are parameter files to the helm template
+	ParameterFiles []HelmParameter `json:"parameterFiles,omitempty" protobuf:"bytes,5,opt,name=parameterFiles"`
```

Comment:
should this be string rather than HelmParameter?

Reply:
I don't think so. Multiple of these should be permitted in the same fashion as helm's --set and --set-string flags:

> --set stringArray            set values on the command line (can specify multiple or separate values with commas: key1=val1,key2=val2)
      --set-file stringArray       set values from respective files specified via the command line (can specify multiple or separate values with commas: key1=path1,key2=path2)
      --set-string stringArray     set STRING values on the command line (can specify multiple or separate values with commas: key1=val1,key2=val2)

Reply:
Best I can figure is that I need to do something similar to https://github.com/tomcruise81/argo-cd/blob/feature/helm-set-file/reposerver/repository/repository.go#L244-L264 in order to handle relative pathing.

Reply:
A file can only be a path, `HelmParameter` defines a value that maybe a `code` (e.g. a boolean) so. So you need to only to use `string` here.

Reply:
So it's not just a string; it's an array of key/value pairs. Each key maps to a path.